script.js

```diff
@@ -109,18 +109,21 @@ function drawCarouselSlideLeft() {
         step = slider.length - 1;
     }
 };
-function changesButtonsStatusDisable(disable) {
-    if (disable === true) {
-        buttonRight.disabled = true;
-        buttonLeft.disabled = true;
-    }
-    else {
-        buttonRight.disabled = false;
-        buttonLeft.disabled = false;
+
+function throttle (callback, limit) {
+    var waiting = false;                     
+    return function () {                                     
+         if (!waiting) {                   
+            callback.apply(this, arguments);  
+            waiting = true;                   
+            setTimeout(function () {          
+                waiting = false;             
+            }, limit);
+        }
```

Comment:
Can you explain what this function does?

Reply:
This is a wrapper for the function so that it is executed with a delay (the function will be executed no more than once in the specified period - limit, even if it is called many times during this period).